exporter/opentelemetry-exporter-otlp-proto-common/src/opentelemetry/exporter/otlp/proto/common/metrics_encoder/__init__.py

```diff
@@ -0,0 +1,237 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import logging
+from typing import Sequence, Any, Mapping, Dict, Callable, List
+
+from opentelemetry.proto.collector.metrics.v1.metrics_service_pb2 import (
+    ExportMetricsServiceRequest,
+)
+from opentelemetry.proto.common.v1.common_pb2 import (
+    KeyValue,
+    InstrumentationScope,
+    AnyValue,
+    ArrayValue,
+    KeyValueList,
+)
+from opentelemetry.proto.metrics.v1 import metrics_pb2 as pb2
+from opentelemetry.sdk.metrics._internal.point import (
+    MetricsData,
+    Gauge,
+    Histogram,
+    Sum,
+)
+from opentelemetry.proto.resource.v1.resource_pb2 import (
+    Resource as PB2Resource,
+)
+from opentelemetry.sdk.resources import Resource
+
+_logger = logging.getLogger(__name__)
+
+
+def _translate_attributes(attributes) -> Sequence[KeyValue]:
```

Comment:
Isn't this equivalent to `_encode_attributes` in common?

Reply:
Nope sadly not, because `_translate_value` is different meaning `_translate_key_values` is different. All part of this discussion https://github.com/open-telemetry/opentelemetry-python/pull/3169#discussion_r1101188361 basically attributes either have Mapping or don't

Reply:
Should we address this now or address it separately @lzchen ?

Reply:
https://github.com/open-telemetry/opentelemetry-python/issues/3202

Reply:
Is the issue an indication that you want to address this later @lzchen? It would actually be less code to address it now

Reply:
@jack-burridge-cfh 
If you could include it easily into this current Pr I have no issues with it :)

Reply:
@lzchen Done was quite a minor commit